client/src/game/carrier.js

```diff
@@ -206,6 +210,30 @@ class Carrier extends EventEmitter {
      this.graphics_colour.visible = true
      if (this.text_garrison) this.text_garrison.visible = true
      this.updateVisibility()
+
+     let clampedScaling = this.userSettings.map.objectsScaling == 'clamped'
```

Comment:
Set a `this.clampedScaling` variable in the `setup` function instead of checking this every time the `onTick` runs.